pydantic/types.py

```diff
@@ -436,6 +437,12 @@ def constr(
     return _registered(type('ConstrainedStrValue', (ConstrainedStr,), namespace))
 
 
+class StrLimited:
+    @classmethod
+    def __class_getitem__(cls, max_length: int) -> Type[ConstrainedStr]:
+        return constr(max_length=max_length, strip_whitespace=True)
```

Comment:
This usage of `__class_getitem__` isn't compatible with typing, at least mypy will give an error for `StrLimited[10]`, and I believe the other major type checkers would too. Mypy gives: `error: The type "Type[StrLimited]" is not generic and not indexable [misc]`. I'd suggest redesigning this to make `StrLimited` an object that implements `__get_item__` instead.